dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,300 @@
+        secretForm.setServiceKey(serviceKey);
+        secretForm.setHostId(host.getIdentifier());
+        secretForm.setParams(paramMap);
+        final Response createSecretResponse = serviceIntegrationResource.createServiceIntegrationSecrets(request, response, secretForm);
+        Assert.assertEquals(HttpStatus.SC_OK, createSecretResponse.getStatus());
+
+        final Response hostIntegrationsResponse = serviceIntegrationResource.getServiceIntegrationByKey(request, response, serviceKey);
+        Assert.assertEquals(HttpStatus.SC_OK, hostIntegrationsResponse.getStatus());
+        final ResponseEntityView responseEntityView2 = (ResponseEntityView) hostIntegrationsResponse.getEntity();
+        final ServiceIntegrationHostView serviceIntegrationHostView2 = (ServiceIntegrationHostView) responseEntityView2.getEntity();
+        final ServiceIntegrationView serviceIntegrationView2 = serviceIntegrationHostView2.getService();
+        Assert.assertEquals(1, serviceIntegrationView2.getConfigurationsCount());
+        Assert.assertEquals("lola", serviceIntegrationView2.getName());
+        final List<HostView> hosts = serviceIntegrationHostView2.getHosts();
+        Assert.assertNotNull(hosts);
+        Assert.assertFalse(hosts.isEmpty());
+        Assert.assertEquals(hosts.get(0).getHostId(),host.getIdentifier());
+        Assert.assertTrue(
+                hosts.stream().anyMatch(hostView -> host.getIdentifier().equals(hostView.getHostId()))
+                );
+
+        final Response detailedIntegrationResponse = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_OK, detailedIntegrationResponse.getStatus());
+        final ResponseEntityView responseEntityView3 = (ResponseEntityView) detailedIntegrationResponse.getEntity();
+        final ServiceIntegrationDetailedView serviceIntegrationDetailedView = (ServiceIntegrationDetailedView) responseEntityView3.getEntity();
+        Assert.assertEquals(serviceIntegrationDetailedView.getHost().getHostId(),host.getIdentifier());
+
+        final Response deleteIntegrationsResponse = serviceIntegrationResource.deleteAllServiceIntegrationSecrets(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_OK, deleteIntegrationsResponse.getStatus());
+
+        final Response detailedIntegrationResponseAfterDelete = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_NOT_FOUND, detailedIntegrationResponseAfterDelete.getStatus());
+
+    }
+
+    @Test
+    public void  Test_Create_service_descriptor_Then_Create_Service_Integration_Then_Delete_One_Single_Secret() throws IOException {
+        final Map<String, Param> paramMap = ImmutableMap.of(
+                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint"),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "param1" appears 4 times in this file; the first occurrence is on line 179](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)